src/operator/controllers/intents_controller.go

```diff
@@ -116,6 +117,7 @@ func NewIntentsReconciler(
 	if enforcementConfig.EnableDatabasePolicy {
 		databaseReconciler := database.NewDatabaseReconciler(client, scheme, otterizeClient)
 		intentsReconciler.group.AddToGroup(databaseReconciler)
+		intentsReconciler.DatabaseReconciler = databaseReconciler
```

Comment:
It's bad practice to change an object from outside. But I'm not sure why this needs to go in the intents reconciler? The database reconciler should handle this all on its own.